main.py

```diff
@@ -37,60 +37,53 @@ def menu(sftp):
                     print("Please input the file path.")
 
             case "ls":
-                # check if command had the -l flag for local
-                if(commandLen > 2):
-                    if("-l" in command):
-                        command = command.remove("-l")
-                        commandLen = len(command)
-                        if(commandLen > 1):
-                            client.printLocalDirectory(command[1])
-                        else:
-                            client.printLocalDirectory(".")
-                    elif("-r" in command):
-                        command = command.remove("-r")
-                        commandLen = len(command)
-                        if(commandLen > 1):
-                            client.printRemoteDirectory(command[1])
-                        else:
-                            client.printRemoteDirectory(".")
-                else:
-                    # remove the flags from the string
-                    if(command is not None and "-l" in command):
-                        command = command.remove("-l")
-                    if(command is not None and "-r" in command):
-                        command = command.remove("-r")
+                flag = None
+                if ("-l" in command): 
+                    command.remove("-l")
+                    flag = "-l"
+                elif ("-r" in command):
+                    command.remove("-r")
+                    flag = "-r"
 
-                    if(command is None):
-                        commandLen = 1
-                    else:
-                        commandLen = len(command)
-                        
-                    # if there is a path after removing the flags
-                    if(commandLen > 1):
-                        client.printLocalDirectory(command[1]) 
-                    # otherwise default to printing the local current dir
-                    else:
-                        client.printLocalDirectory(".")
+                commandLen = len(command)
+                if flag == "-l":
+                    match commandLen:
+                        case 2:
+                            client.printLocalDirectory(sftp, command[1])
```

Comment:
Shouldn't need to pass sftp to printLocalDirectory

Reply:
Good point. I'll fix that in a separate branch after merging these two.

Reply:
Wait, that's actually dumb, I'll fix it in this branch. For some reason I was under the impression the function itself took sftp and it was a problem with the function, not that we're accidentally passing 2 arguments when it only needs 1.